applications/ParticleMechanicsApplication/python_scripts/mpm_explicit_solver.py

```diff
@@ -0,0 +1,119 @@
+
+    @classmethod
+    def GetDefaultSettings(cls):
+        this_defaults = KratosMultiphysics.Parameters("""{
+            "time_integration_method"   : "explicit",
+            "scheme_type"   : "forward_euler",
+            "stress_update" : "USL"
+        }""")
+        this_defaults.AddMissingParameters(super(MPMExplicitSolver, cls).GetDefaultSettings())
+        return this_defaults
+
+
+    def AddVariables(self):
+        super(MPMExplicitSolver, self).AddVariables()
+        self._AddDynamicVariables(self.grid_model_part)
+        grid_model_part = self.GetGridModelPart()
+
+        # Adding explicit variables
+        grid_model_part.AddNodalSolutionStepVariable(KratosMultiphysics.FORCE_RESIDUAL)
+        grid_model_part.AddNodalSolutionStepVariable(KratosMultiphysics.RESIDUAL_VECTOR)
+
+        scheme_type = self.settings["scheme_type"].GetString()
+        KratosMultiphysics.Logger.PrintInfo("::[MPMExplicitSolver]:: ", "Variables are all added.")
+
+    ### Protected functions ###
+
+    def _CreateSolutionScheme(self):
+        grid_model_part = self.GetGridModelPart()
+        domain_size = self._GetDomainSize()
+        block_size  = domain_size
+        if (self.settings["pressure_dofs"].GetBool()):
+            block_size += 1
+
+        # Check whether compressibility is considered
+        isCompressible = self.settings["compressible"].GetBool()
+        grid_model_part.ProcessInfo.SetValue(KratosParticle.IS_COMPRESSIBLE, isCompressible)
+
+        # Setting the time integration schemes
+        scheme_type = self.settings["scheme_type"].GetString()
+        isCentralDifference = False
+        StressUpdateOption = 0
+
+        if(scheme_type == "forward_euler" or scheme_type == "Forward_Euler"):
+            stress_update = self.settings["stress_update"].GetString() #0 = USF, 1 = USL, 2 = MUSL
+            if(stress_update == "USF" or stress_update == "usf"):
+                StressUpdateOption = 0
+            elif(stress_update == "USL" or stress_update == "usl"):
+                StressUpdateOption = 1
+            elif(stress_update == "MUSL" or stress_update == "musl"):
+                StressUpdateOption = 2
+            else:
+                err_msg = "The requested stress update \"" + stress_update + "\" is not available!\n"
+                err_msg += "Available options are: \"USF\", \"USL\",\"MUSL\""
+            grid_model_part.ProcessInfo.SetValue(KratosParticle.EXPLICIT_STRESS_UPDATE_OPTION, StressUpdateOption)
+            grid_model_part.ProcessInfo.SetValue(KratosParticle.IS_EXPLICIT_CENTRAL_DIFFERENCE, False)
+        elif(scheme_type == "central_difference" or scheme_type == "Central_Difference"):
+            isCentralDifference = True
+            grid_model_part.ProcessInfo.SetValue(KratosParticle.EXPLICIT_STRESS_UPDATE_OPTION, 0)
+            grid_model_part.ProcessInfo.SetValue(KratosParticle.IS_EXPLICIT_CENTRAL_DIFFERENCE, True)
```

Comment:
I think this is usually done through the Scheme

Reply:
I mean, what time integration one uses

Reply:
A difference here is that this one scheme is doing 4 variants of explicit time integration, while usually one scheme does one form of time integration only. Since the 4 variants I'm doing are very similar, I put them in one scheme with the logic drivers EXPLICIT_STRESS_UPDATE_OPTION and IS_EXPLICIT_CENTRAL_DIFFERENCE. The VARIABLE IS_EXPLICIT_CENTRAL_DIFFERENCE also has to be accessible from the explicit utilities cpp, which is why I stuck with VARIABLES accessible from the processinfo. Although EXPLICIT_STRESS_UPDATE_OPTION  could be removed (and passed to the scheme constructor to be a scheme member variable) I kept it as a VARIABLE to be consistent with IS_EXPLICIT_CENTRAL_DIFFERENCE.

Reply:
For now I think it is fine but imagine someone adds another option, then using the true/false for the IS_EXPLICIT_CENTRAL_DIFFERENCE would already be no longer sufficient

Not saying you should change it now, but I think it will become necessary in the future